pkg/tuned/controller.go

```diff
@@ -450,6 +439,35 @@ func providerExtract(provider string) error {
 	return nil
 }
 
+// Read the Cloud Provider name from openshiftTunedProvider file and
+// extract/write 'provider' only if the file does not exist or it does not
+// match 'provider'.  Returns indication whether the 'provider' changed and
+// an error if any.
+func providerSync(provider string) (bool, error) {
+	var providerCurrent string
+
+	f, err := os.Open(openshiftTunedProvider)
+	if err != nil {
+		if os.IsNotExist(err) {
+			// The provider file hasn't been created yet.
+			return len(provider) > 0, providerExtract(provider)
+		}
+		return false, err
+	}
+	defer f.Close()
+
+	var scanner = bufio.NewScanner(f)
+	for scanner.Scan() {
+		providerCurrent = strings.TrimSpace(scanner.Text())
+	}
```

Comment:
On second thought *IF* we own `openshiftTunedProvider` would it make sense to just use `os.ReadFile` to read its (IIUC?) single-word content?

Reply:
Yes, will change this, thank you.

Reply:
I'd write as 
```
func providerSync(provider string) (bool, error) {
        providerCurrent, err := os.ReadFile(openshiftTunedProvider)
        if err != nil {
                if os.IsNotExist(err) {
                        return len(provider) > 0, providerExtract(provider)
                }
                return false, err
        }

        if provider == string(providerCurrent) {
                return false, nil            
        }

        return true, providerExtract(provider)
}
```

but besides that, yep, looks nice